subsys/net/lib/sockets/sockets.c

```diff
@@ -2499,6 +2515,48 @@ int zsock_getsockopt_ctx(struct net_context *ctx, int level, int optname,
 		case TCP_NODELAY:
 			ret = net_tcp_get_option(ctx, TCP_OPT_NODELAY, optval, optlen);
 			return ret;
+
+		case TCP_KEEPIDLE:
+			if (IS_ENABLED(CONFIG_NET_TCP_KEEPALIVE)) {
+				ret = net_tcp_get_option(ctx, TCP_OPT_KEEPIDLE,
+							 optval, optlen);
+				if (ret < 0) {
+					errno = -ret;
+					return -1;
+				}
+
+				return 0;
+			}
+
+			break;
+
+		case TCP_KEEPINTVL:
+			if (IS_ENABLED(CONFIG_NET_TCP_KEEPALIVE)) {
+				ret = net_tcp_get_option(ctx, TCP_OPT_KEEPINTVL,
+							 optval, optlen);
+				if (ret < 0) {
+					errno = -ret;
+					return -1;
+				}
+
+				return 0;
+			}
+
+			break;
+
+		case TCP_KEEPCNT:
+			if (IS_ENABLED(CONFIG_NET_TCP_KEEPALIVE)) {
+				ret = net_tcp_get_option(ctx, TCP_OPT_KEEPCNT,
+							 optval, optlen);
+				if (ret < 0) {
+					errno = -ret;
+					return -1;
+				}
+
+				return 0;
+			}
+
+			break;
```

Comment:
All these options are nearly identical, so could we do it like this:
```
case TCP_KEEPIDLE:
	__fallthrough;
case TCP_KEEPINTVL:
	__fallthrough;
case TCP_KEEPCNT:
			if (IS_ENABLED(CONFIG_NET_TCP_KEEPALIVE)) {
				ret = net_tcp_get_option(ctx, get_tcp_option(optname),
							 optval, optlen);
				if (ret < 0) {
					errno = -ret;
					return -1;
				}

				return 0;
			}
```

and the `get_tcp_option(optname)` could return the correct option according to option name.

Reply:
Good point - I've optimized option set/get as suggested.